packages/arkor-local/python/mlx/train_shim.py

```
@@ -0,0 +1,306 @@
+
+    lora_mod.run(args, training_callback=callback)
+
+    # Flush any checkpoint saved after the last loss report, then publish
+    # the final adapter under a stable, normalised path.
+    callback.publish_new_checkpoints()
+    final_dir = _publish_adapter(raw_dir, adapters_dir / "final")
+    emit({"type": "completed", "adapterDir": str(final_dir)})
+    return 0
+
+
+def _resolve_iterations(train_cfg: dict, train_count: int):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_resolve_iterations`

(ANN202)